functions/Restart-DbaService.ps1

```diff
@@ -115,17 +115,22 @@ function Restart-DbaService {
     end {
         $processArray = [array]($processArray | Where-Object { (!$InstanceName -or $_.InstanceName -in $InstanceName) -and (!$Type -or $_.ServiceType -in $Type) })
         foreach ($service in $processArray) {
-            if ($Force -and $service.ServiceType -eq 'Engine' -and !($processArray | Where-Object { $_.ServiceType -eq 'Agent' -and $_.InstanceName -eq $service.InstanceName -and $_.ComputerName -eq $service.ComputerName })) {
-                Write-Message -Level Verbose -Message "Adding Agent service to the list for service $($service.ServiceName) on $($service.ComputerName), since -Force has been specified"
-                #Construct parameters to call Get-DbaService
-                $serviceParams = @{
-                    ComputerName = $service.ComputerName
-                    InstanceName = $service.InstanceName
-                    Type         = 'Agent'
+            if ($Force -and $service.ServiceType -eq 'Engine') {
+                foreach ($dependentService in @("Agent", "PolyBase", "Launchpad")) {
```

Comment:
Get-DbaService supports returning multiple types at once, I'm pretty sure it would be faster to get all three types initially and then add missing services, as opposed to doing three requests - one for each service.

Reply:
@nvarscar Thanks for the review! I didn't consider the multiple types going in.
I have kept the loop, to prevent duplicates from being added to the $processArray, but it is now just building a collection of $dependentService. Now $dependentService is being used for the single call to Get-DbaService